src/compute.geometry/IO/Schema.cs

```diff
@@ -61,4 +67,215 @@ public ResthopperObject(object obj)
             this.Type = obj.GetType().FullName;
         }
     }
+
+    public abstract class ResthopperParam
+    {
+        public string Name { get; set; } = "";
+        public string Label { get; set; }
+        public GHTypeCodes TypeCode { get; set; }
+        public IGH_Param Param { get; set; }
+
+        public bool TryParseName(string label, out string message)
+        {
+            message = "";
+            Label = label;
+
+            // Begin parsing label for type information
+            var labelData = label.Split(':');
+
+            if (labelData.Length < 2)
+            {
+                message = "Invalid group name format.";
+                return false;
+            }
+
+            if (!int.TryParse(labelData[1], out var val))
+            {
+                message = "Could not parse integer value from group name.";
+                return false;
+            }
+
+            try
+            {
+                TypeCode = (GHTypeCodes)val;
+            }
+            catch
+            {
+                message = "No type found for integer value provided. It may be incorrect or unsupported.";
+                return false;
+            }
+
+            if (labelData.Length == 3)
+            {
+                Name = labelData[2];
+            }
+
+            return true;
+        }
+    }
+
+    public class ResthopperInput : ResthopperParam
+    {
+        public static Dictionary<GHTypeCodes, string> TypeCodeToParamType { get; } = new Dictionary<GHTypeCodes, string>()
```

Comment:
if we got picky with naming conventions, these dictionaries could get cut out and we could just make minor alterations to the `GHTypeCode` name